### PR TITLE
ODIM-3390 add http/rpc handler for delete volume

### PR DESCRIPTION
**Pull request by irappa-pattar**
- Added HTTP handler for delete volume
- Added RPC handler for delete volume

### CODE REVIEW

**Review by a-ajith — APPROVED** [review]

**Review by skhan-anjum — APPROVED** [review]

File: svc-api/router/router.go [hunk 1]
```diff
@@ -252,6 +253,7 @@ func Router() *iris.Application {
 	storage.Get("/{rid}/Drives", system.GetSystemResource)
 	storage.Get("/{rid}/Drives/{rid2}", system.GetSystemResource)
 	storage.Post("/{rid}/Volumes", system.CreateVolume)
+	storage.Delete("/{rid}/Volumes/{rid2}", system.DeleteVolume)
```
Comment (jeevan-kamkar) [inline svc-api/router/router.go]: change this to id2 and rid

**Review by jeevan-kamkar — CHANGES_REQUESTED** [review]
Address the change

File: svc-api/router/router.go (hunk 1 above)
Comment (irappa-pattar) [reply svc-api/router/router.go]: changed

**Review by jeevan-kamkar — APPROVED** [review]
LGTM

File: svc-systems/systems/storage.go [hunk 2]
```diff
@@ -254,10 +254,108 @@ func searchItem(slice []string, val string) bool {
...
+	err := json.Unmarshal(req.RequestBody, &volume)
+	if err != nil {
+		errorMessage := "Error while unmarshaling the create volume request: " + err.Error()
+		log.Printf(errorMessage)
+		resp = common.GeneralError(http.StatusBadRequest, response.MalformedJSON, errorMessage, []interface{}{}, nil)
+		return resp
+	}
+
+	// spliting the uuid and system id
+	requestData := strings.Split(req.SystemID, ":")
```
Comment (shivacharanms) [inline svc-systems/systems/storage.go:274]: Scenarios for "uuid:" where there is no system id present should be handled as the len(requestData) will be 2

Comment (shivacharanms) [reply svc-systems/systems/storage.go:274]: Add UT for the same

**Review by shivacharanms — CHANGES_REQUESTED** [review]

File: svc-systems/systems/storage.go (hunk 2 above)
Comment (irappa-pattar) [reply svc-systems/systems/storage.go:274]: Handled the scenario and added test case for that

File: svc-systems/systems/storage.go [hunk 3]
```diff
@@ -254,10 +254,108 @@ func searchItem(slice []string, val string) bool {
...
+
+	if strings.EqualFold(plugin.PreferredAuthType, "XAuthToken") {
+		var err error
+		contactRequest.HTTPMethodType = http.MethodPost
+		contactRequest.DeviceInfo = map[string]interface{}{
+			"UserName": plugin.Username,
+			"Password": string(plugin.Password),
+		}
+		contactRequest.OID = "/ODIM/v1/Sessions"
+		_, token, getResponse, err := scommon.ContactPlugin(contactRequest, "error while creating session with the plugin: ")
```
Comment (shivacharanms) [inline svc-systems/systems/storage.go:327]: mocking plugin status scenario to be handled

Comment (irappa-pattar) [reply svc-systems/systems/storage.go:327]: Handled

Comment (irappa-pattar) [reply svc-systems/systems/storage.go:327]: handled

**Review by shivacharanms — APPROVED** [review]

**Pull request merged**